### PR TITLE
Handle nested lists for UNWIND in compiled runtime

### PR DESCRIPTION
**Pull request by pontusmelke**
Fix a bug where we weren't properly handling types in nested lists.

### CODE REVIEW

File: enterprise/cypher/cypher-compiled-runtime-3.2/src/main/scala/org/neo4j/cypher/internal/compiled_runtime/v3_2/codegen/LogicalPlanConverter.scala [hunk 1]
```diff
@@ -526,6 +526,10 @@ object LogicalPlanConverter {
         case CypherCodeGenType(symbols.ListType(innerType), ListReferenceType(innerReprType))
           if RepresentationType.isPrimitive(innerReprType) =>
           (CypherCodeGenType(innerType, innerReprType), UnwindPrimitiveCollection(opName, collection))
+        case CypherCodeGenType(symbols.ListType(innerType), ListReferenceType(innerReprType)) =>
+          (CypherCodeGenType(innerType, innerReprType), ir.UnwindCollection(opName, collection))
+        case CypherCodeGenType(symbols.ListType(innerType), _) =>
```
Comment (sherfert) [inline enterprise/cypher/cypher-compiled-runtime-3.2/src/main/scala/org/neo4j/cypher/internal/compiled_runtime/v3_2/codegen/LogicalPlanConverter.scala]: Now that case is duplicate... that is probably not intentional?

Comment (pontusmelke) [reply enterprise/cypher/cypher-compiled-runtime-3.2/src/main/scala/org/neo4j/cypher/internal/compiled_runtime/v3_2/codegen/LogicalPlanConverter.scala]: No not quite, in the case where the inner type is primitive we use `UnwindPrimitiveCollection` and in the other we use ` ir.UnwindCollection`

Comment (sherfert) [reply enterprise/cypher/cypher-compiled-runtime-3.2/src/main/scala/org/neo4j/cypher/internal/compiled_runtime/v3_2/codegen/LogicalPlanConverter.scala]: Sorry, I should have been more specific. Lines 531 and 533.

Comment (pontusmelke) [reply enterprise/cypher/cypher-compiled-runtime-3.2/src/main/scala/org/neo4j/cypher/internal/compiled_runtime/v3_2/codegen/LogicalPlanConverter.scala]: yes you are right, that was a mistake :)

**Review by sherfert — APPROVED** [review]

**Pull request merged**